modules/ballerina-core/src/main/java/org/ballerinalang/util/semantics/SemanticAnalyzer.java

```diff
@@ -1202,6 +1253,10 @@ public void visit(AssignStmt assignStmt) {
 
         visitSingleValueExpr(rExpr);
         BType rhsType = rExpr.getType();
+        if (assignStmt.isVarDeclaration()) {
+            assignVariableRefTypes(new Expression[]{lExpr}, new BType[]{rhsType});
```

Comment:
Do you really need to invoke 'assignVariableRefTypes' method here? :) 

I would just set the type then and there